chain/near/src/chain.rs

```diff
@@ -235,7 +245,10 @@ impl TriggersAdapterTrait<Chain> for TriggersAdapter {
             .map(|r| NearTrigger::Receipt(Arc::new(r)))
             .collect();
 
-        trigger_data.push(NearTrigger::Block(shared_block.cheap_clone()));
+        // TODO: Confirm this is correct
+        if block_filter.trigger_every_block {
```

Comment:
@maoueh @leoyvens does this look correct to you?

Should we always have a trigger even if we have no block handlers?

Reply:
It looks right to me, the trigger should only be added if there are block handlers.

Reply:
```data_source.mapping.block_handlers.is_empty``` is the only condition that sets trigger_every_block to true, so I don't think it could trigger without a block handler (this may be the same thing that @leoyvens is saying right before I hit 'Comment')

Reply:
All comments about filtering up in https://github.com/graphprotocol/graph-node/pull/3372/files#r830173700 applies here also.

Reply:
https://github.com/graphprotocol/graph-node/pull/3372/files#diff-99488584ebb0c0295142cf3018e8097c0cd6b5bdf74964b23b1d7fa23d42ba8aR112 this is why i was asking. Before we were using source being set as filter rather than block handlers existing. And we would also set this to true in that case. 

Now it's exactly as you describe, it will only be true if there is a block handler